### ISSUE TITLE
[Model Request] Hojo-ASR-Multi-V1

### ISSUE THREAD

**Issue by AppleSheeple** [CrispStrobe/CrispASR#438]
https://huggingface.co/HojoAI/Hojo-ASR-Multi-V1

Apache 2.0 License

**Comment by CrispStrobe** [comment]
Thanks — and this one is more tractable than most model requests, because we
already have both halves.

Licence confirmed **Apache-2.0** through the HF API rather than read off the
card.

From `config.yaml`, it is an Encoder-Adapter-LLM stack:

```yaml
model:
  encoder_path: "Qwen3-Omni-30B-A3B-Instruct"
  llama_path:   "Qwen3-4B-Instruct-2507"
  whisper_path: "whisper-large-v3"
  linear_units: 640
  num_blocks:   2
  input_layer:  "linear"
```

So: the **Qwen3-Omni audio encoder**, a 2-block linear adapter, and a
**Qwen3-4B** decoder — with whisper-large-v3 supplying the feature extractor.

The reason that matters: `src/moss_transcribe.cpp` already implements the stock
Qwen3-Omni-MoE audio encoder (conv2d stem + 32 pre-LN layers), we ship Qwen3 LLM
runtimes, and the whisper front end is the oldest thing in the repo. The adapter
is two linear blocks. That is a very different proposition from #436's X-ASR and
Dolphin, which each need a whole new encoder family (zipformer, WeNet U2++).

Two things I would want to check before promising anything, because they are
where this kind of port actually goes wrong:

1. **Which encoder layers are used, and whether the checkpoint carries its own
   copy.** The repo ships `merged_full_model.safetensors` plus a bare
   `Qwen3-Omni-30B-A3B-Instruct/config.json` — if the encoder weights are inside
   the merged file we are fine; if it expects to pull a 30B checkpoint at load
   time, the practical model size is very different from what the file listing
   suggests.
2. **Whether the adapter does multi-frame stacking.** The card describes a
   "customized multi-frame acoustic fusion structure", and `linear_units: 640`
   against a 1280-wide encoder output hints at a 2:1 stack. Getting that
   ordering wrong produces fluent, confident, wrong transcripts — the failure
   mode that is hardest to notice.

Languages tagged are de/fr/it/pt/es. No promises on timing, but this is on the
list as a realistic port rather than a research project.

**Comment by CrispStrobe** [comment]
Hojo-ASR-Multi-V1 is on `main` now as `--backend hojo-asr`. `-m hojo-asr` downloads [`cstr/Hojo-ASR-Multi-V1-GGUF`](https://huggingface.co/cstr/Hojo-ASR-Multi-V1-GGUF), which has a Q4_K default (4.4 GB) and F16.

Architecture: the Qwen3-Omni audio tower (32 layers), a 2-block WeNet Conformer adapter and a Qwen3-4B language model.

Checked stage by stage against the upstream package on German and French clips at F16:

| stage (worst-frame cos) | German | French |
| --- | --- | --- |
| mel spectrogram | 1.000000 | 1.000000 |
| encoder output | 0.999998 | 0.999999 |
| adapter output | 0.999994 | 0.999995 |
| prefill embeddings | 0.999997 | 0.999997 |
| first-step logits | 1.000000 | 1.000000 |

End to end, the greedy transcript is identical to upstream's greedy output on both clips at F16, and at Q4_K on French. At Q4_K on German one word differs.

Decoding is greedy by default. `config.yaml` names beam 4, which is available with `-bs 4`, but CrispASR's beam ranking doesn't yet reproduce `transformers`' `generate()`.

**Comment by AppleSheeple** [comment]
Thanks.

Three notes:
* The backend has no entry in the library backend list (It does in the CLI).
* I successfully converted and tested [Hojo-ASR-V1](https://huggingface.co/HojoAI/Hojo-ASR-V1) model, and it does work differently/better with English input. So perhaps a separate model card for that is useful.
* I don't know if this is a model limitation, but testing the above on a long(-ish) chunk, and putting the transcription quality aside, transcription output appears to be short (test sample [tt.wav.zip](https://github.com/user-attachments/files/32574766/tt.wav.zip)). Manual chunking helps, but avoiding that would be preferable.

**Comment by AppleSheeple** [comment]
> The backend has no entry in the library backend list (It does in the CLI).

This got fixed.
